py/frequenz/api/electricity_trading/v1/_types.py

```diff
@@ -0,0 +1,1187 @@
+
+        Returns:
+            DeliveryArea object corresponding to the protobuf message.
+        """
+        return cls(code=delivery_area.code, code_type=delivery_area.code_type)
+
+    def to_pb(self) -> delivery_area_pb2.DeliveryArea:
+        """Convert a DeliveryArea object to protobuf DeliveryArea.
+
+        Returns:
+            Protobuf message corresponding to the DeliveryArea object.
+        """
+        return delivery_area_pb2.DeliveryArea(code=self.code, code_type=self.code_type)
+
+
+class DeliveryDuration(enum.Enum):
+    """
+    Specifies the time increment, in minutes, used for electricity deliveries and trading. These durations serve as the basis for defining the delivery period in contracts, and they dictate how energy is scheduled and delivered to meet contractual obligations.
+    """
+
+    UNSPECIFIED = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_UNSPECIFIED
+    """Default value, indicates that the duration is unspecified."""
+
+    MINUTES_5 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_5
+    """5-minute duration."""
+
+    MINUTES_15 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_15
+    """15-minute contract duration."""
+
+    MINUTES_30 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_30
+    """30-minute contract duration."""
+
+    MINUTES_60 = delivery_duration_pb2.DeliveryDuration.DELIVERY_DURATION_60
+    """1-hour contract duration."""
+
+    @classmethod
+    def from_pb(
+        cls, delivery_duration: delivery_duration_pb2.DeliveryDuration.ValueType
+    ) -> DeliveryDuration:
+        """Convert a protobuf DeliveryDuration value to DeliveryDuration enum.
+
+        Args:
+            delivery_duration: Delivery duration to convert.
+
+        Returns:
+            Enum value corresponding to the protobuf message.
+        """
+        if not any(e.value == delivery_duration for e in cls):
+            _logger.warning(
+                "Unknown delivery duration %s. Returning UNSPECIFIED.",
+                delivery_duration,
+            )
+            return cls.UNSPECIFIED
+
+        return cls(delivery_duration)
+
+    def to_pb(self) -> delivery_duration_pb2.DeliveryDuration.ValueType:
+        """Convert a DeliveryDuration object to protobuf DeliveryDuration.
+
+        Returns:
+            Protobuf message corresponding to the DeliveryDuration object.
+        """
+        return delivery_duration_pb2.DeliveryDuration.ValueType(self.value)
+
+
+@dataclass(frozen=True)
+class DeliveryPeriod:
+    """Time period during which the contract is delivered. It is defined by a start timestamp and a duration.
+
+    Attributes:
+        start: Start UTC timestamp represents the beginning of the delivery period.
+            This timestamp is inclusive, meaning that the delivery period starts
+            from this point in time.
+        duration: The length of the delivery period.
+    """
+
+    start: datetime
+    duration: DeliveryDuration
+
```

Comment:
Could be a good idea here to check if the start and duration fields are the correct type. Easier for bug catching in the future. We could add a post_init function here.

```python
    def __post_init__(self):
        if not isinstance(self.start, datetime):
            raise ValueError("start must be a datetime object")
```